File: java/BubbleSort.java
package java;
import java.util.Arrays;

public class BubbleSort {
  public static void main(String[] args) {
    int[] array = { 6, 4, 5, 3, 1, 2 };
    print(Arrays.toString(array));
    for (int i = 0, len = array.length; i < len - 1; i++) {
      for (int j = 0; j < len - 1 - i; j++) {
        if (array[j] > array[j+1]) {
          int temp = array[j];
          array[j] = array[j+1];
          array[j+1] = temp;
        }
      }
    }
    print(Arrays.toString(array));
  }

  private static void print(String string) {
    System.out.println(string);
  }
}

File: javascript/bubbleSort.js
const array = [6, 4, 5, 3, 2, 1];
console.log(array);
for (let i = 0, len = array.length; i < array.length - 1; i++) {
  for (let j = 0; j < len - 1 - i; j++) {
    if (array[j] > array[j+1]) {
      const temp = array[j];
      array[j] = array[j+1];
      array[j+1] = temp;
    }
  }
}
console.log(array);
